src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -649,87 +635,105 @@ private void AddTransactionToWallet(Transaction transaction, TxOut utxo, int? bl
                 return;
             }
 
-            // If the details of this spending transaction are seen for the first time.
             if (spentTransaction.SpendingDetails == null)
             {
+                // 1) If no existing spending details, always set new spending details.
```

Comment:
@codingupastorm @quantumagi pardon me, my knowledge is still vague in these areas. Can you explain why or who a spent transaction's spending details can be null?

Reply:
This is the case when the transaction is in the wallet, aka we "own" it, it's just sitting there ready to be spent and contributes to our balance.

Reply:
Ah makes sense!